test/test_transforms_v2_refactored.py

```diff
@@ -3881,3 +3899,741 @@ def test_correctness_perspective_bounding_boxes(self, startpoints, endpoints, fo
+        check_functional(F.equalize, make_input())
+
+    @pytest.mark.parametrize(
+        ("kernel", "input_type"),
+        [
+            (F.equalize_image, torch.Tensor),
+            (F._equalize_image_pil, PIL.Image.Image),
+            (F.equalize_image, tv_tensors.Image),
+            (F.equalize_video, tv_tensors.Video),
+        ],
+    )
+    def test_functional_signature(self, kernel, input_type):
+        check_functional_kernel_signature_match(F.equalize, kernel=kernel, input_type=input_type)
+
+    @pytest.mark.parametrize(
+        "make_input",
+        [make_image_tensor, make_image_pil, make_image, make_video],
+    )
+    def test_transform(self, make_input):
+        check_transform(transforms.RandomEqualize(p=1), make_input())
+
+    # We are not using the default `make_image` here since that uniformly samples the values over the whole value range.
+    # Since the whole point of F.equalize is to transform an arbitrary distribution of values into a uniform one,
+    # the information gain is low if we already provide something really close to the expected value.
+    def _make_correctness_image(self, *, type, **kwargs):
+        shape = (3, 117, 253)
+        dtype = torch.uint8
+        device = "cpu"
+
+        max_value = get_max_value(dtype)
+
+        def make_constant_image(*, value_factor=0.0):
+            return torch.full(shape, value_factor * max_value, dtype=dtype, device=device)
+
+        def make_uniform_band_distributed_image(*, low_factor=0.1, high_factor=0.9):
+            return torch.testing.make_tensor(
+                shape, dtype=dtype, device=device, low=low_factor * max_value, high=high_factor * max_value
+            )
+
+        def make_beta_distributed_image(*, alpha=2.0, beta=5.0):
+            image = torch.distributions.Beta(alpha, beta).sample(shape)
+            image.mul_(get_max_value(dtype)).round_()
+            return image.to(dtype=dtype, device=device)
+
+        make_fn = {
+            "constant": make_constant_image,
+            "uniform_band_distributed": make_uniform_band_distributed_image,
+            "beta_distributed": make_beta_distributed_image,
+        }[type]
+        return tv_tensors.Image(make_fn(**kwargs))
+
+    @pytest.mark.parametrize(
+        "make_correctness_image_kwargs",
+        [
+            *[dict(type="constant", value_factor=value_factor) for value_factor in [0.0, 0.5, 1.0]],
+            *[
+                dict(type="uniform_band_distributed", low_factor=low_factor, high_factor=high_factor)
+                for low_factor, high_factor in [
+                    (0.0, 0.25),
+                    (0.25, 0.75),
+                    (0.75, 1.0),
+                ]
+            ],
+            *[
+                dict(type="beta_distributed", alpha=alpha, beta=beta)
+                for alpha, beta in [
+                    (0.5, 0.5),
+                    (2.0, 2.0),
+                    (2.0, 5.0),
+                    (5.0, 2.0),
```

Comment:
Is there value in testing every single one of these?

E.g. do we really need to test beta-distributed input? And if so, do we need to test across all these beta/alpha tuples? Similarly do we need to check all `low_factor, high_factor` tuples and all `value_factor` values? Do we even need to test `type="constant"` on top of `type="uniform_band_distributed"`?

Reply:
> do we really need to test beta-distributed input?

The intention here was to have not only uniformly sampled inputs.

> And if so, do we need to test across all these beta/alpha tuples?

Probably not. One is likely enough to surface bugs in case there are some.

> Similarly do we need to check all `low_factor, high_factor` tuples and all `value_factor` values? 

Probably not.

> Do we even need to test `type="constant"` on top of `type="uniform_band_distributed"`?

Constant images are definitely an edge case for equalize, but they are very different from uniform band images.

---

The `F.equalize_image` kernel is quite involved. Our intention of using many different input images was to make sure we didn't mess up the algorithm, since the implementation is very different from v1 and PIL. Now that we made sure that indeed our implementation is solid, we can probably get away with testing the edge cases (constant images) and one or two others.